File: Python/GasMileage/gas_mileage.py
#! /usr/bin/env python3
# This program calculates the mileage of a car

# define some variables
total_gallons_used = 0
total_miles_driven = 0

gallons_used = float( input( "Enter the gallons used ( -1 to end ): " ) ) # Prompt user for input

# Use a loop to get miles driven and calculate miles per gallon
while gallons_used != -1:
    miles_driven = int( input( "Enter the miles driven: " ) ) # Prompt user for input

    # Calculate and print the miles per gallon
    miles_per_gallon = miles_driven / gallons_used
    print( "The miles / gallon for this tank was {:.6f}".format( miles_per_gallon ) )

    total_gallons_used += gallons_used # Add gallons_used to total_gallons_used
    total_miles_driven += miles_driven # Add miles_driven to total_miles_driven

    gallons_used = float( input( "Enter the gallons used ( -1 to end ): " ) ) # Prompt user for input

# Calculate and print the total mpg
if total_gallons_used > 0:
    total_miles_per_gallon = total_miles_driven / total_gallons_used
    print( "The overall average miles / gallon was {:.6f}".format( total_miles_per_gallon ) )
else:
    print( "Nothing was inputted." )
    print( "The overall average miles / gallon is 0" )

File: C++/GasMileage/gasMileage.cpp
// This program calculates a car's gas mileage
#include <iostream>
#include <iomanip>

// Program execution begins at function main
int main()
{
    // Declare variables
    int milesDriven;
    int gallonsUsed;
    double milesPerGallon;
    int totalMilesDriven = 0;
    int totalGallonsUsed = 0;
    double totalMilesPerGallon;

    std::cout << "Enter miles driven ( -1 to quit ): "; // Prompt user for input
    std::cin >> milesDriven; // Read miles driven from user

    // Use a loop to get gallons used and calculate mpg
    while ( milesDriven != -1 )
    {
        std::cout << "Enter gallons used: "; // Prompt user for input
        std::cin >> gallonsUsed; // Read gallonsUsed from user

        // Calculate and print mpg
        milesPerGallon = static_cast< double >( milesDriven ) / gallonsUsed;
        std::cout << "MPG this tankful: " << std::setprecision( 6 ) << std::fixed
            << milesPerGallon << std::endl;

        totalGallonsUsed += gallonsUsed; // Increment totalGallonsUsed by gallonsUsed
        totalMilesDriven += milesDriven; // Increment totalMilesDriven by milesDriven

        // Check if totalMilesDriven is greater than 0
        if ( totalGallonsUsed > 0 )
        {
            // Calculate and print total mpg
            totalMilesPerGallon = static_cast< double >( totalMilesDriven ) / totalGallonsUsed;
            std::cout << "Total MPG: " << std::setprecision( 6 ) << std::fixed
                << totalMilesPerGallon << std::endl;
        } // end if
        else
        {
            std::cerr << "Nothing was inputted.\n";
            std::cerr << "Total MPG is 0" << std::endl;
        } // end else

        std::cout << "\nEnter miles driven ( -1 to quit ): "; // Prompt user for input
        std::cin >> milesDriven; // Read miles driven from user
    } // end while

    return 0; // Indicate program terminated successfully
} // end function
